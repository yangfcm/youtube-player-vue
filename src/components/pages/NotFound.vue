<template>
  <div>
    <app-blank></app-blank>
    <div class="ui red message">
      <app-blank></app-blank>
      <div class="ui header aligned center">The page you requested doesn't exist.</div>
      <app-blank></app-blank>
      <div class="ui header aligned center">
        <router-link to="/" class="ui red basic button">Back to Home</router-link>
      </div>
      <app-blank></app-blank>
    </div>
  </div>
</template>

<script>
import Blank from "../common/Blank";
export default {
  components: {
    appBlank: Blank,
  },
};
</script>

<style scoped></style>
