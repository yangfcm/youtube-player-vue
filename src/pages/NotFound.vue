<script setup lang="ts"></script>

<template>Not found</template>
