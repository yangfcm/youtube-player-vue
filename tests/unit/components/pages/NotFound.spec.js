import { shallowMount, RouterLinkStub } from "@vue/test-utils";
import NotFound from "@/components/pages/NotFound";

describe("Test not found page", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(NotFound, {
      stubs: {
        RouterLink: RouterLinkStub,
      },
    });
  });
  afterEach(() => {
    wrapper.destroy();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have back to home button", () => {
    expect(wrapper.find("a").props().to).toBe("/");
  });
});
