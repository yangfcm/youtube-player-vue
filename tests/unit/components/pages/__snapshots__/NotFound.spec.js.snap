// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test not found page should match snapshot 1`] = `
<div>
  <app-blank-stub></app-blank-stub>
  <div class="ui red message">
    <app-blank-stub></app-blank-stub>
    <div class="ui header aligned center">The page you requested doesn't exist.</div>
    <app-blank-stub></app-blank-stub>
    <div class="ui header aligned center"><a class="ui red basic button">Back to Home</a></div>
    <app-blank-stub></app-blank-stub>
  </div>
</div>
`;
